tik_splitter/download/downloader.py

```diff
@@ -27,80 +29,146 @@ class VideoDownloader(Downloader):
     def __init__(self, output_path: Path = VIDEO_PATH):
         super().__init__(output_path)
 
-    def download_video(self, url: str) -> Video | None:
+    def download_video(self, url: str, start_time: str = None, end_time: str = None) -> Video | None:
+        if (start_time is None and end_time is not None) or (start_time is not None and end_time is None):
+            raise ValueError("If either start time or end time is provided, both must be provided")
         try:
+            output_path = self._output_path
             output_directory = str(self._output_path)  # Convert Path to string
-            output_directory_path = Path(output_directory)
-            output_directory_path.mkdir(parents=True, exist_ok=True)
 
             youtube = YouTube(url)
+            video_title = clean_string(youtube.title)  # regex video name
+            video_length = youtube.length
             video_stream = youtube.streams.get_highest_resolution()
 
             self._logger.info(f"Downloading YouTube video: {youtube.title}")
-            original_filename = video_stream.download(output_directory)
-            new_filename = original_filename.replace(" ", "_")
-            new_filepath = output_directory_path / new_filename
-            Path(original_filename).replace(new_filepath)
+            video_download = video_stream.download(output_directory)
+            new_filename = video_title + ".mp4"
+            video_filepath = output_path / new_filename
+
+            if start_time and end_time:  # TODO: if start_ or end_time:
```

Comment:
Need a follow-up on these TODOs to have default values for start and end time if the other is given in the input params